code/modules/projectiles/gun.dm

```diff
@@ -543,11 +549,28 @@
 			for(var/obj/item/grab/G in M.grabbed_by)
 				grabstate = max(grabstate, G.state)
 			if(grabstate >= GRAB_NECK)
-				damage_mult = 2.5
+				accuracy_mult = 4
 			else if(grabstate >= GRAB_AGGRESSIVE)
-				damage_mult = 1.5
-	P.multiply_projectile_damage(damage_mult)
+				accuracy_mult = 8
+
+	P.multiply_projectile_accuracy(accuracy_mult)
+
+/obj/item/gun/proc/grabbed_inaccuracy(obj/item/projectile/P, mob/user) //TODO: make length of gun be a disadvantage
```

Comment:
Without the barrel length nerf, rifles will be weirdly good during CQC Grabs. In fact I'm pretty sure this makes point blanking someone's head off with a rifle better than using a proper melee weapon.

Reply:
All guns are close to guaranteed to hit someone grabbed even before this buff. 
The buff itself is hardly significant however, the disadvantage on close range accuracy is identical to the advantage from aggressive grabs. (*4 increases leftmost bit by 3)
This is a pretty strong nerf to cqc-magdump as is, so holding off on merging just gives more power to magdumpers.